### PR TITLE
Load delayed libraries only once

### PR DESCRIPTION
**Pull request by tmathern**
## Issue

Fix multiple loading of delayed libraries.

## 🔗 Test URLs:

- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/7
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/7
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/1
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/1
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/2
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/2
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/3
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/3
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/4
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/4
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/5
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/5
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/6
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/6
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/8
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/8
- Before: https://main--sgws-creative--hlxsites.hlx.page/bloominbrands/p/9
- After: https://load-libs-once--sgws-creative--hlxsites.hlx.page/bloominbrands/p/9

## 📝 Description:

- Currently, having fragments on a page can lead to a library being loaded multiple times (once per fragment relying on it), as multiple fragments may bring in the request to load the same delayed library, or multiple fragments may reuse libraries. This is to prevent multiple loads of the same library on a page.
- Improve fragments loading in vertical selector.
- Make more event listeners passive.

### CODE REVIEW

File: blocks/chart-map/chart-map.js [hunk 1]
```diff
@@ -158,8 +158,6 @@ function drawMap(block, mapHolder, mapData, theme) {
 
 let echartsLoaded = false;
 export default function decorate(block) {
-  window.hasCharts = true;
```
Comment (tmathern) [inline blocks/chart-map/chart-map.js:161]: Not needed anymore.

File: scripts/delayed.js [hunk 2]
```diff
@@ -18,15 +26,28 @@ const loadScript = (url, attrs) => {
...
   }
   head.append(script);
+
   return script;
 };
 
 // add more delayed functionality here
-// Maps (projections)
+// Maps (and projections)
 if (document.querySelector('div.chart-map-container')) {
```
Comment (tmathern) [inline scripts/delayed.js:35]: Also ensures we have the needed d3 dependency before loading echarts, so maps always get the correct projection.

File: blocks/vertical-selector/vertical-selector.js [hunk 3]
```diff
@@ -8,11 +8,12 @@ import {
 } from '../../scripts/scripts.js';
 import { decorateIcons } from '../../scripts/lib-franklin.js';
 
-async function loadFragment(cell) {
+/* async */ function loadFragment(cell) {
```
Comment (tmathern) [inline blocks/vertical-selector/vertical-selector.js:11]: "Symbolically" async, since we're letting it lazy-load.

File: scripts/delayed.js [hunk 4]
```diff
@@ -9,7 +9,15 @@ sampleRUM('cwv');
 // loads additional delayed scripts
 const loadScript = (url, attrs) => {
   const head = document.querySelector('head');
+  const loadedScripts = head.querySelectorAll('script');
+  const loadedScriptsElements = [...loadedScripts];
+  for (let i = 0; i < loadedScriptsElements.length; i += 1) {
+    const currentScript = loadedScriptsElements[i];
+    if (currentScript.src === url) return null;
```
Comment (tmathern) [inline scripts/delayed.js:16]: If script URL already loaded, do not reload it.

File: scripts/delayed.js [hunk 5]
```diff
@@ -18,20 +26,34 @@ const loadScript = (url, attrs) => {
...
+  const d3MapUtils = loadScript('https://cdnjs.cloudflare.com/ajax/libs/d3-geo/1.9.1/d3-geo.min.js', {
     type: 'text/javascript',
   });
+  if (d3MapUtils !== null) {
+    d3MapUtils.onload = () => {
+      const echarts = loadScript(echartsLink, {
+        type: 'text/javascript',
+      });
+      if (echarts !== null) {
+        echarts.onload = () => {
```
Comment (tmathern) [inline scripts/delayed.js:46]: Make sure the dependency is there before echarts get here, so everything is set up properly for maps (the projections for maps come from d3).

**Review by badvision — APPROVED** [review]

**Pull request merged**